menu_item.h

```diff
@@ -30,6 +30,7 @@ struct MenuItem {
   virtual void CursorDown();
   virtual void CursorLeft();
   virtual void CursorRight();
+  virtual void OnX();
```

Comment:
Nit: In the `MenuItem` I think it's useful to have the functions describe the action rather than the button. Maybe `OnX` should be something like `OnActivateAll` or `OnActivateGroup`?

That might also clarify the expected behavior as well.

- If I invoked "OnActivateAll" on a suite, I'd expect it to run through all of the tests inside of the suite.
- If I invoked it on a test, I think I'd just expect it to invoke the test.

Reply:
Thinking more - I guess maybe if I'm looking at the list of tests it might be more usable to have it also activate the whole group as well even if it feels a bit strange to ignore the cursor position (and a bit odd that at the root level it'll run everything inside of the selected item).

Reply:
I was thinking of it as an OnActivateCurrentSuite button.  

I'm inclined to sometimes cherry pick a single test then when I see that it looks good I might want to kick them all off (even while i'm already in the results screen) to make sure the others didn't regress.    

From that standpoint a run entire suite button is nice since you wouldn't have to back out two screens to run all, or back out one screen and think you have to take the cursor to the top to make sure you don't miss any so it won't miss any.

I think it has good usability (if you know it exists) my only thought was also pushing a run all menu item to the test list for each suite.

Reply:
Sounds reasonable to me. I think the "run all" would be just one more item to scroll past when you're trying to run a specific test. Since this is geared more towards developers than end users I think we're fine just mentioning the X button behavior in the README.md (maybe add that to this PR?).

Reply:
done.  I also added a deactivate() when we call it from a test results screen so it will pop us up a menu to the test list when complete.  Without it you didn't know it finished and it sort of looked hung.